examples/typescript/platform-ref-aws/crossplane.ts

```diff
@@ -0,0 +1,175 @@
+  config.addProvider('registry.upbound.io/crossplane/provider-helm', '>=v0.3.6-0');
+
+  return yaml;
+}
+
+export function postgresYaml(crossplanePackage: App): Chart {
+  const yaml = new Chart(crossplanePackage, 'postgres-api');
+  const xrd = new crossplane.CompositeResourceDefinition(yaml, 'postgres-xrd', {
+    name: 'compositepostgresqlinstances.aws.platformref.crossplane.io',
+  });
+
+  xrd.group('aws.platformref.crossplane.io');
+  xrd.claimKind('PostgreSQLInstance').plural('postgresqlinstances');
+  xrd.kind('CompositePostgreSQLInstance').plural('compositepostgresqlinstances');
+  xrd.connectionSecret().key('username').key('password').key('endpoint').key('port').defaultNamespace('crossplane-system');
+
+  let xrdNetRef: SchemaPropString;
+  let xrdStorageGB: SchemaPropInteger;
+  xrd.version('v1alpha1').served().referencable()
+    .spec().with(Prop.for({ object: (spec) => {
+      spec.propObject('parameters').required().with(Prop.for({ object: (params) => {
+        params.uiSection( { 
+          title: 'Database Size', 
+          description: 'Enter information to size your database'});
+
+        xrdStorageGB = params.propInteger('storageGB').required().min(1).max(500)
+          .description('GB of storage for your database')
+          .uiInput({
+            name: 'storageGB',
+            inputType: InputType.SINGLE_INPUT,
+            title: 'Storage (GB)',
+            default: 5
+          });
+
+        params.propObject('networkRef').required()
+          .description('A reference to the Network object that this postgres should be connected to.')
+          .with(Prop.for({ object: (networkRef) => {
+
+            xrdNetRef = networkRef.propString('id')
+              .description('ID of the Network object this ref points to.')
+              .required()
+              .uiInput({
+                name: 'networkRef',
+                inputType: InputType.SINGLE_INPUT,
+                title: 'Network Ref',
+                default:'platform-ref-aws-network',
+                customError: 'Network ref is required and should match the network ref of the app cluster.'
+              });
+
+          }}));
+
+          params.propString('writeSecretRef').implicit().required()
+          .uiInput({
+            name: 'writeSecretRef',
+            inputType: InputType.SINGLE_INPUT,
+            title: 'Connection Secret Ref',
+            description: 'name of the secret to write to this namespace',
+            default: 'db-conn',
+            path: '.spec.writeConnectionSecretToRef.name'
+          })
+      }}));
+
+    }}));
+
+  const composition = new crossplane.Composition(yaml, 'postgres-composition', xrd, {
+    name: 'compositepostgresqlinstances.aws.platformref.crossplane.io',
+    metadata: {
+        labels: {
+          provider: 'aws',
+        },
+      },
+  });
+
+  //TODO: use imported provider-aws types
+  composition.addResource(db.DbSubnetGroup.propsWithGVK({
```

Comment:
@eladb Splitting out `propsWithGVK()` is in https://github.com/awslabs/cdk8s/pull/447 will let us compose imported L1 resources with `kind` and `apiVersion` as template resources for `Composition` resources.